### PR TITLE
Setup tests and initial set of dependencies

### LINKED ISSUE #1: Setup project
Setup initial project structure and get an initial test running which prints a report (e.g. as pdf).

This is maybe the starting ground for how to set up the dependencies.

Later on also test interaction with quarkus-jaxb.

See also: https://github.com/quarkiverse/quarkiverse/wiki/Checklist#after-the-repository-is-created

```
 <dependency>
            <groupId>xerces</groupId>
            <artifactId>xercesImpl</artifactId>
            <version>${version.xerces}</version>
            <!-- Runtime so this dependency does not conflict with jaxb-maven-plugin. Dependency is only needed for the jasper print. -->
            <scope>runtime</scope>
            <exclusions>
                <exclusion>
                    <groupId>xml-apis</groupId>
                    <artifactId>xml-apis</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>xalan</groupId>
            <artifactId>xalan</artifactId>
            <version>${version.xalan}</version>
            <!-- Runtime so this dependency does not conflict with jaxb-maven-plugin. Dependency is only needed for the jasper print. -->
            <scope>runtime</scope>
            <exclusions>
                <exclusion>
                    <groupId>xml-apis</groupId>
                    <artifactId>xml-apis</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>net.sf.jasperreports</groupId>
            <artifactId>jasperreports</artifactId>
            <version>${version.jasperreports}</version>
        </dependency>
        <dependency>
            <groupId>net.sf.jasperreports</groupId>
            <artifactId>jasperreports-functions</artifactId>
            <version>${version.jasperreports.functions}</version>
        </dependency>
        <dependency>
            <groupId>net.sf.jasperreports</groupId>
            <artifactId>jasperreports-fonts</artifactId>
            <version>${version.jasperreports.fonts}</version>
        </dependency>
        <!-- Needed when printing barcodes with barbeque-api for e.g. labels -->
        <dependency>
            <groupId>net.sourceforge.barbecue</groupId>
            <scope>runtime</scope>
            <artifactId>barbecue</artifactId>
            <version>${version.barbecue}</version>
        </dependency>
        <!-- Needed when printing barcodes for barcode4j-api e.g. labels (QRCode)-->
        <dependency>
            <groupId>net.sf.barcode4j</groupId>
            <scope>runtime</scope>
            <artifactId>barcode4j</artifactId>
            <version>${version.barcode4j}</version>
            <exclusions>
                <exclusion>
                    <groupId>xml-apis</groupId>
                    <artifactId>xml-apis</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>org.apache.ant</groupId>
                    <artifactId>ant</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>commons-cli</groupId>
                    <artifactId>commons-cli</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>avalon-framework</groupId>
                    <artifactId>avalon-framework</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
        <dependency>
            <groupId>com.google.zxing</groupId>
            <scope>runtime</scope>
            <artifactId>core</artifactId>
            <version>${version.zxing}</version>
        </dependency>
        <dependency>
            <groupId>org.apache.xmlgraphics</groupId>
            <scope>runtime</scope>
            <artifactId>batik-bridge</artifactId>
            <version>${version.batik-bridge}</version>
            <exclusions>
                <exclusion>
                    <groupId>xml-apis</groupId>
                    <artifactId>xml-apis</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>xml-apis-ext</groupId>
                    <artifactId>xml-apis-ext</artifactId>
                </exclusion>
                <exclusion>
                    <groupId>xalan</groupId>
                    <artifactId>xalan</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
```

### PR DESCRIPTION
**Pull request by Postremus**
fixes #1

### CODE REVIEW

**Comment by Postremus** [comment]
Right now I am facing a funny one:

```
[ERROR] Failed to execute goal com.alexnederlof:jasperreports-plugin:2.8:jasper (default) on project example-report: Execution default 
of goal com.alexnederlof:jasperreports-plugin:2.8:jasper failed: Plugin com.alexnederlof:jasperreports-plugin:2.8 or one of its depende
ncies could not be resolved: Failed to collect dependencies at com.alexnederlof:jasperreports-plugin:jar:2.8 -> net.sf.jasperreports:ja
sperreports:jar:6.15.0 -> com.lowagie:itext:jar:2.1.7.js8: Failed to read artifact descriptor for com.lowagie:itext:jar:2.1.7.js8: Coul
d not transfer artifact com.lowagie:itext:pom:2.1.7.js8 from/to maven-default-http-blocker (http://0.0.0.0/): Blocked mirror for reposi
tories: [jaspersoft-third-party (http://jaspersoft.artifactoryonline.com/jaspersoft/third-party-ce-artifacts/, default, releases+snapshots), jr-ce-releases (http://jaspersoft.jfrog.io/jaspersoft/jr-ce-releases, default, releases+snapshots)] -> [Help 1]
```

mvn clean install on the example-report module.

The plugin I normally use for jasper compilation is not maintained anymore.
https://github.com/alexnederlof/Jasper-report-maven-plugin

Might have to either fork that one, or try out the officical one.

**Comment by Postremus** [comment]
Managed to fix it by forcing an updated jasper version :)

```
<build>
        <plugins>
            <plugin>
                <groupId>com.alexnederlof</groupId>
                <artifactId>jasperreports-plugin</artifactId>
                <version>${jasperreports-plugin.version}</version>
                <executions>
                    <execution>
                        <phase>process-sources</phase>
                        <goals>
                            <goal>jasper</goal>
                        </goals>
                    </execution>
                </executions>
                <dependencies>
                    <dependency>
                        <groupId>net.sf.jasperreports</groupId>
                        <artifactId>jasperreports</artifactId>
                        <version>${jasperreports.version}</version>
                    </dependency>
                </dependencies>
            </plugin>
        </plugins>
    </build>
```

I still asked If I could adopt the maven plugin.

**Comment by Postremus** [comment]
I can use something like this to export the report as zip if need be.
Might need to implement a zipclassloader in that case though..

```
<plugin>
				<groupId>org.apache.maven.plugins</groupId>
				<artifactId>maven-assembly-plugin</artifactId>
				<configuration>
					<descriptors>
						<descriptor>assemblies/zip.xml</descriptor>
					</descriptors>
					<finalName>${project.artifactId}</finalName>
					<appendAssemblyId>false</appendAssemblyId>
					<archiverConfig>
						<compress>false</compress>
					</archiverConfig>
				</configuration>
				<executions>
					<execution>
						<id>create-archive</id>
						<phase>package</phase>
						<goals>
							<goal>single</goal>
						</goals>
					</execution>
				</executions>
			</plugin>
```

assembly.xml

```
<assembly
	xmlns="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.3"
	xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
	xsi:schemaLocation="http://maven.apache.org/plugins/maven-assembly-plugin/assembly/1.1.3 http://maven.apache.org/xsd/assembly-1.1.3.xsd">
	<id>zip</id>
	<formats>
		<format>zip</format>
	</formats>
	<includeBaseDirectory>false</includeBaseDirectory>
	<fileSets>
		<fileSet>
			<directory>target/classes/</directory>
			<outputDirectory>.</outputDirectory>
			<useDefaultExcludes>true</useDefaultExcludes>
			<excludes>
				<exclude>*.zip</exclude>
				<exclude>*.xsd</exclude>
				<exclude>*.jrxml</exclude>
				<exclude>*.xml</exclude>
			</excludes>
		</fileSet>
		<fileSet>
			<directory>target/jasper/</directory>
			<outputDirectory>.</outputDirectory>
			<useDefaultExcludes>true</useDefaultExcludes>
		</fileSet>		
	</fileSets>
</assembly>
```

**Comment by Postremus** [comment]
Waiting until my fork of the maven plugin is released
https://github.com/pro-crafting/Jasper-report-maven-plugin/issues

File: runtime/pom.xml [hunk 1]
```diff
@@ -47,4 +41,17 @@
...
+    <dependency>
+      <groupId>io.quarkus</groupId>
+      <artifactId>quarkus-arc</artifactId>
+    </dependency>
+
+    <dependency>
+      <groupId>net.sf.jasperreports</groupId>
+      <artifactId>jasperreports</artifactId>
+      <version>${jasperreports.version}</version>
+    </dependency>
```
Comment (melloware) [inline runtime/pom.xml:55]: If Jasper includes libs like POI you might want to do an Exclude here and use Quarkus POI which has already been updated for GraalVM Native.  There may be some others as well.

Comment (Postremus) [reply runtime/pom.xml]: done, at least in integration tests. For xlsx files, poi needs to be provided by the user.

**Comment by gastaldi** [comment]
@all-contributors add @Postremus for code and maintenance

**Pull request merged**